tests/test_k8s_manifests.py

```
@@ -0,0 +1,530 @@
+# ---------------------------------------------------------------------------
+# Fixture: render kustomize output once per session
+# ---------------------------------------------------------------------------
+
+K8S_DIR = Path(__file__).parent.parent / "k8s"
+
+
+@pytest.fixture(scope="session")
+def manifests() -> list[dict[str, Any]]:
+    """Render `kubectl kustomize k8s/` and return parsed YAML documents."""
+    try:
+        result = subprocess.run(
```

Ruff (0.15.9)

[error] `subprocess` call: check for execution of untrusted input

(S603)